src/foam/approval/ApprovableAwareDAO.js

```diff
@@ -0,0 +1,450 @@
+foam.CLASS({
+  package: 'foam.approval',
+  name: 'ApprovableAwareDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  javaImports: [
+    'foam.core.X',
+    'foam.core.PropertyInfo',
+    'java.util.Iterator',
+    'foam.dao.DAO',
+    'java.util.Map',
+    'java.util.List',
+    'java.util.Set',
+    'foam.mlang.MLang',
+    'foam.mlang.MLang.*',
+    'foam.core.FObject',
+    'foam.dao.ArraySink',
+    'java.util.ArrayList',
+    'foam.util.SafetyUtil',
+    'foam.nanos.auth.User',
+    'foam.core.Detachable',
+    'foam.dao.AbstractSink',
+    'foam.nanos.logger.Logger',
+    'foam.lib.PropertyPredicate',
+    'foam.nanos.ruler.Operations',
+    'foam.nanos.auth.LifecycleState',
+    'foam.nanos.auth.LifecycleAware',
+    'foam.mlang.predicate.Predicate',
+    'foam.approval.ApprovalStatus',
+    'foam.approval.ApprovalRequest',
+    'foam.comics.v2.userfeedback.UserFeedbackAware',
+    'foam.comics.v2.userfeedback.UserFeedback',
+    'foam.comics.v2.userfeedback.UserFeedbackStatus',
+    'foam.comics.v2.userfeedback.UserFeedbackException',
+    'foam.nanos.crunch.UCJQueryService',
+    'foam.approval.Approvable',
+    'foam.approval.ApprovableAware',
+    'foam.approval.RoleApprovalRequest'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'daoKey'
+    },
+    {
+      class: 'Class',
+      name: 'of'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'sendSingleRequest',
+      type: 'void',
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'req', type: 'RoleApprovalRequest' },
+        { name: 'userId', type: 'long' }
+      ],
+      javaCode: `
+        RoleApprovalRequest request = (RoleApprovalRequest) req.fclone();
+        request.clearId();
+        request.setApprover(userId);
+        ((DAO) x.get("approvalRequestDAO")).put_(x, request);
+      `
+    },
+    {
+      name: 'fullSend',
+      type: 'void',
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'request', type: 'RoleApprovalRequest' },
+        { name: 'obj', type: 'FObject' }
+      ],
+      javaCode:`
+      DAO requestingDAO;
+      Logger logger = (Logger) x.get("logger");
+
+      if ( request.getDaoKey().equals("approvableDAO") ) {
+        DAO approvableDAO = (DAO) x.get("approvableDAO");
+
+        Approvable approvable = (Approvable) approvableDAO.find(request.getObjId());
+
+        requestingDAO = (DAO) x.get(approvable.getDaoKey());
+      } else {
+        requestingDAO = (DAO) x.get(request.getDaoKey());
+      }
+
+      String modelName = requestingDAO.getOf().getObjClass().getSimpleName();
+
+      UCJQueryService ucjQueryService = (UCJQueryService) x.get("ucjQueryService");
```

Comment:
add UCJQueryService to FOAM services file.

Reply:
As discussed before we abstracted it to use UserQueryService and added that to the foam services file